### ISSUE TITLE
White towers dissappear if emulated console is reset after game is started

### ISSUE THREAD

**Issue by deadair3210** [PCSX2/pcsx2#1752]
***PCSX2 version:***
``(For example, stable build v1.4.0 (or) development builds v1.5.xxx)``
dev 1.5-1774
- **Note**: It's preferred to only report bugs which are present on the [latest development builds](http://pcsx2.net/download/development.html).

- **Note2**: The new backwards compatibility (psx) mode is in active development. Issue reports are too early at this point.

***PCSX2 options:***
``(For example, Clamping/Rounding, Speedhacks, GameFixes)``
Memcard Folder, Automatically manage saves unchecked
- **Note**: Generally it's recommended to have settings as default as possible and use the ``Safest`` preset along with MTVU speed hack disabled. If there's no issue with these settings but there is an issue with some other non-default option, then please state it explicitly.

***Plugins used:***
``(Relevant/Non-default settings of GS/PAD/SPU2/CDVD/USB plugins)``

- **Note**: In case the issue is a graphical problem with the GSDX plugin, please provide basic information like ``Renderer/Internal resolution/CRC Hack level/Other relevant options`` which were used to reproduce the issue. Additionally, providing a [GS dump](http://forums.pcsx2.net/Thread-How-to-create-a-proper-GS-dump) of the issue would also be helpful.

***Description of the issue:***
``(What's the problem? - Screenshots showing the issue if applicable)``
Not quite sure if this is a bug or not, but I don't remember the towers on the BIOS of the PS2 ever disappearing unless you deleted the file, but in pcsx2, the towers will not be in their normal spot if you reboot the emulated console after a game has been loaded

***How to reproduce the issue:***
``(Detailed step by step guide on how to replicate the issue)``
Load BIOS with a memcard that has some saves already on it so that the towers will show up
Start a game and wait until at least after the BIOS releases control (After the PS2LOGO)
Restart the system in any way other than shutting down the console first
Towers will be missing until the console is shutdown and started again

***Last known version to work:***
``(If the issue isn't present in an older PCSX2 version, it would be helpful to mention the full title of that version. Narrowing it down to the build/commit that first introduced the issue is even better!)``
Unknown

***PC specifications:***
``(For example: CPU: Intel Core 2 Duo E4600 GPU: Gigabyte GTX 750Ti OS: Windows 10 64 Bit)``
i5 2500K @ 4.0 GHz
Raedon HD 7870
14 GB RAM
Windows 10 64-bit

**Comment by refractionpcsx2** [comment]
does it only happen on folder memory cards or does it happen on standard memorycard files too?

**Comment by deadair3210** [comment]
After some testing, I think I can say that it only happens with Folder Memory Cards

**Comment by refractionpcsx2** [comment]
That kind of makes sense due to how they work.  From what I understand, the folder memorycard rearranges itself so only the game you are running appears on the memorycard, that way you get 8mb of data for every game, so it's possible it could confuse the crap out of the bios :p

**Comment by deadair3210** [comment]
Shouldn't that be turned off when the user has Automatically manage game saves turned off? If it is a normal thing, could it be modified to always have the System file mapped in to the memory card so that settings and the towers can be used like a normal console?

**Comment by refractionpcsx2** [comment]
No idea, I've never touched the code :P I just have the option selected and leave it at that lol

**Comment by deadair3210** [comment]
I would look at the code myself and see, but I have only just started learning C++ and Javascript. I usually have the filter turned off because I value the nostalgic original operation of the BIOS over maybe having some games stop on a file every once in a while, lol

**Comment by MrCK1** [comment]
I use folder memory cards myself, but I can't reproduce this bug with the steps listed above.

**Comment by MrCK1** [comment]
Correcting my original comment, I've been able to duplicate this. As said above, it only occurs with folder memory cards.

**Comment by RedDevilus** [comment]
This issue is odd. If i use a 64MB it will see this (already has formatted and saves in it)
![image](https://user-images.githubusercontent.com/24227051/98749218-785ab880-23bb-11eb-9ceb-6c7b662f99d3.png)
If i do PCSX2 and shutdown it will give another state. If you do Boot Iso and click reset it just replicates previous behavior.
It looks consistent in 8,16 and 32 MB but the size and amount are never correct on any type, it's just that 64 MB and folder memcard amplify that issue by showing mostly no tower. This is 64MB:
![image](https://user-images.githubusercontent.com/24227051/98749592-45fd8b00-23bc-11eb-8f84-722316ceeae4.png)
64MB also only shows 1 tower or even none.

This is what the OPs issue is:
![image](https://user-images.githubusercontent.com/24227051/98749731-8d841700-23bc-11eb-9103-09ec32798c2c.png)

But it does seem that disabling managing saves gives the same behavior as 8,16 and 32MB:
![image](https://user-images.githubusercontent.com/24227051/98749875-e8b60980-23bc-11eb-9a4d-ea08dd41a4a3.png)

TL;DR They are all broken to an extend compared to what i have on my PS2, 64 MB and folder memcard have it even worse.

**Comment by Masamune3210** [comment]
Maybe the bios does some checking when loading the file? Considering how old this bug is it might just be a byproduct of how folder MC's work

**Comment by RedDevilus** [comment]
Yeah it's just odd that it sometimes it does show 3 towers on occasion even with the option enabled although harder to trigger.

**Comment by Masamune3210** [comment]
Well back when I made this issue I had no way to do much research, I'm getting a network adapter for my PS2 soon again so I might do some research

**Comment by Mrlinkwii** [comment]
is this still an issue on master ?

**Comment by RedDevilus** [comment]
If anyone want to see all the towers, here is a fried folder memcard:
[FormattedRedMcd001_converted.ps2.zip](https://github.com/PCSX2/pcsx2/files/12241869/FormattedRedMcd001_converted.ps2.zip)

**Comment by Noxillio** [comment]
I'm unsure if this is relevant at all but I use the default 8MB memory card files and am now encountering history resets. Additionally, and I'm unsure if this is noteworthy either, but when PCSX2 writes to the history files it fucks the dates all up. I'm unsure if it's just me or whether it's because I use turbo but regardless to my knowledge the console will only write to the history file when launching a game in order to increment the launch count and set the date. And since PCSX2 sets the clock to whatever the system clock is, this shouldn't be happening. Please correct me if I am mistaken about any of this.

**Comment by Mrlinkwii** [comment]
>And since PCSX2 sets the clock to whatever the system clock is, this shouldn't be happening.

not quite true , since #12208 you can set anytime you wish on a per game setting ,

**Comment by Noxillio** [comment]
> you can set anytime you wish on a per game setting

That’s good to know for those who need it, I wasn’t aware that was added.

**Comment by Noxillio** [comment]
I'm still seeing this issue. Haven't emulated in a while but after the past week of gameplay all my towers got wiped out again.

(still using an 8MB memory card file per defaults)

**Comment by Masamune3210** [comment]
The towers work fine for me, even through the 2.0 Browser update. As long as the history file is present on a memory card inserted into the emulated console at boot and is valid, they should work just as they would on an actual console. To note, the towers don't actually use the date for anything, it's just for bookkeeping, they only respond to total amount of times the game was launched and the count is updated right before PS2LOGO is decrypted and shown before control is passed to the game.
